packages/@eventual/core-runtime/src/local/stores/bucket-store.ts

```diff
@@ -0,0 +1,220 @@
+import {
+  Bucket,
+  CopyBucketObjectOptions,
+  CopyBucketObjectResponse,
+  DurationSchedule,
+  GetBucketObjectOptions,
+  GetBucketObjectResponse,
+  ListBucketRequest,
+  ListBucketResult,
+  PutBucketObjectResponse,
+} from "@eventual/core";
+import crypto from "crypto";
+import { Readable, Stream } from "stream";
+import { BucketStore } from "../../stores/bucket-store.js";
+import { LocalEnvConnector } from "../local-container.js";
+import { paginateItems } from "./pagination.js";
+
+export interface LocalBucketStoreProps {
+  localConnector: LocalEnvConnector;
+}
+
+export class LocalBucketStore implements BucketStore {
+  private objects: Record<
+    string,
+    Record<
+      string,
+      {
+        body: Uint8Array | string | Buffer;
+        contentLength: number;
+        etag: string;
+      }
+    >
+  > = {};
+
+  // TODO: publish events for the stream
+  constructor(private props: LocalBucketStoreProps) {}
+  async get(
+    bucketName: string,
+    key: string,
+    options?: GetBucketObjectOptions
+  ): Promise<GetBucketObjectResponse | undefined> {
+    const bucket = this.objects[bucketName];
```

Comment:
i assume we may want to store these on the file system one day

Reply:
Likely. There is a larger story to persist local data across restarts.